integration/management-api-tests/src/test/java/org/wso2/micro/integrator/api/RegistryResourcesTestCase.java

```diff
@@ -0,0 +1,1450 @@
+/*
+ *Copyright (c) 2022, WSO2 LLC. (http://www.wso2.com) All Rights Reserved.
+ *
+ *WSO2 Inc. licenses this file to you under the Apache License,
+ *Version 2.0 (the "License"); you may not use this file except
+ *in compliance with the License.
+ *You may obtain a copy of the License at
+ *
+ *http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *Unless required by applicable law or agreed to in writing,
+ *software distributed under the License is distributed on an
+ *"AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ *KIND, either express or implied.  See the License for the
+ *specific language governing permissions and limitations
+ *under the License.
+ */
+
+package org.wso2.micro.integrator.api;
+
+import org.apache.http.HttpResponse;
+import org.awaitility.Awaitility;
+import org.json.JSONObject;
+import org.testng.Assert;
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+import org.wso2.esb.integration.common.utils.ESBIntegrationTest;
+import org.wso2.esb.integration.common.utils.ESBTestConstant;
+import org.wso2.esb.integration.common.utils.clients.SimpleHttpClient;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.TimeUnit;
+
+import org.skyscreamer.jsonassert.JSONAssert;
+import org.wso2.esb.integration.common.utils.common.TestConfigurationProvider;
+
+public class RegistryResourcesTestCase extends ESBIntegrationTest {
+
+    private String accessToken;
+
+    @BeforeClass(alwaysRun = true)
+    public void setEnvironment() throws Exception {
+        super.init();
+    }
+
+    @Test(groups = { "wso2.esb" }, priority = 2, description = "Test adding registry content (text)")
+    public void testRegistryPostText() throws IOException {
+
+        if (!isManagementApiAvailable) {
+            Awaitility.await().pollInterval(50, TimeUnit.MILLISECONDS).atMost(DEFAULT_TIMEOUT, TimeUnit.SECONDS)
+                    .until(isManagementApiAvailable());
+        }
+
+        String endpoint = "https://" + hostName + ":" + (DEFAULT_INTERNAL_API_HTTPS_PORT + portOffset) + "/management/"
+                + "registry-resources/content";
+        String registryPath = "registry/config/testFolder/test-text.txt";
+        String queryParameters = "?path=" + registryPath
+                + "&mediaType=testMediaType";
+        File file = new File(TestConfigurationProvider.getResourceLocation(ESBTestConstant.ESB_PRODUCT_GROUP)
+                + "/registry-resources/test-initial.txt");
```

Comment:
Using `/` not work with windows environment. Shall we use `File.pathSeparator` instead?

Reply:
I used formatPath() method in the management API to resolve this